core/src/main/scala/pink/cozydev/protosearch/LastTermRewrite.scala

```diff
@@ -16,31 +16,15 @@
 
 package pink.cozydev.protosearch
 
-import cats.data.NonEmptyList
 import pink.cozydev.lucille.Query
-import pink.cozydev.lucille.Query.{TermQ, FieldQ, PrefixTerm, Group, OrQ}
+import pink.cozydev.lucille.TermQuery
 
 object LastTermRewrite {
 
-  def termToPrefix(q: Query): Query =
-    q match {
-      case TermQ(t) =>
-        Group(NonEmptyList.one(OrQ(NonEmptyList.of(TermQ(t), PrefixTerm(t)))))
-      case FieldQ(fn, TermQ(t)) =>
-        Group(
-          NonEmptyList.one(OrQ(NonEmptyList.of(FieldQ(fn, TermQ(t)), FieldQ(fn, PrefixTerm(t)))))
-        )
-      case _ => q
-    }
+  def termToPrefix(tq: TermQuery): Query = tq match {
+    case Query.Term(t) =>
+      Query.Group(Query.Or(Query.Term(t), Query.Prefix(t)))
```

Comment:
I feel like I understood this when you first wrote it and now I'm confused. Why is the last term rewritten to a `Group` of an `Or` instead of just returning the `Or` query directly?

Reply:
Funny... I don't exactly know why I did it that way. A raw `Or` would work perfectly fine here.

I feel like I mentally picture it with brackets, so I reached for group.

Reply:
> I feel like I mentally picture it with brackets, so I reached for group.

This makes a lot of sense though 😆